### ISSUE TITLE
Beachmat not finding H5Cpp.h during install

### ISSUE THREAD

**Issue by cypranowska** [tatami-inc/beachmat#6]
I've been getting this error when attempting to install Beachmat from Bioconductor: 

```
Bioconductor version 3.6 (BiocInstaller 1.28.0), ?biocLite for help
* installing *source* package 'beachmat' ...
** libs
g++ -std=gnu++11 -I/global/home/groups/consultsw/sl-7.x86_64/modules/r/3.4.3/lib64/R/include -DNDEBUG  -I"/global/home/groups/fc_nmjreg/rpak/Rcpp/include" -I"/global/home/groups/fc_nmjreg/rpak/Rhdf5lib/include" -I/usr/local/include   -fpic  -g -O2 -c HDF5_utils.cpp -o HDF5_utils.o
In file included from HDF5_utils.h:4:0,
                 from HDF5_utils.cpp:1:
beachmat.h:17:23: fatal error: c++/H5Cpp.h: No such file or directory
 #include "c++/H5Cpp.h"
                       ^
compilation terminated.
```

Not sure if the issue is with beachmat or Rhdf5lib, but if I run `ls -l /global/home/groups/fc_nmjreg/rpak/Rhdf5lib/include` H5Cpp.R is there. 

Here's my sessionInfo if it's of any help:
```
R version 3.4.3 (2017-11-30)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS: /global/home/groups/consultsw/sl-7.x86_64/modules/r/3.4.3/lib64/R/lib/libRblas.so
LAPACK: /global/home/groups/consultsw/sl-7.x86_64/modules/r/3.4.3/lib64/R/lib/libRlapack.so

locale:
[1] C

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets
[8] methods   base

other attached packages:
 [1] HDF5Array_1.7.8      DelayedArray_0.5.21  BiocParallel_1.13.1
 [4] IRanges_2.13.26      S4Vectors_0.17.33    BiocGenerics_0.25.3
 [7] matrixStats_0.53.1   Rcpp_0.12.15         rhdf5_2.23.5
[10] Rhdf5lib_1.1.5       BiocInstaller_1.28.0

loaded via a namespace (and not attached):
[1] compiler_3.4.3 tools_3.4.3
```

**Comment by LTLA** [comment]
Looking at your session information reveals that you're mixing BioC-devel versions of packages (middle number of odd) with BioC-release versions (middle number is even). In particular, all of your packages other than _BiocInstaller_ are from the BioC-devel branch. This is mostly likely responsible for the error, as the BioC-devel version of _Rhdf5lib_ was updated to export its headers directly to `<PATH>/include/` rather than `<PATH>/include/c++`. As a result, the BioC-_release_ version of _beachmat_ fails to find the headers. This would not be a problem with the BioC-devel version of _beachmat_, obtainable on R-devel with version 1.29.* of _BiocInstaller_; nor if you were using the BioC-release version of _Rhdf5lib_.

In short, read https://bioconductor.org/install/#why-biocLite.

**Comment by cypranowska** [comment]
Wow! I can't believe I totally missed that. Thank you for pointing out my mistake.
